app_dart/lib/src/service/scheduler.dart

```diff
@@ -450,7 +454,33 @@ class Scheduler {
             success = true;
           }
         } else {
-          await luciBuildService.rescheduleUsingCheckRunEvent(checkRunEvent);
+          final String sha = checkRunEvent.checkRun!.headSha!;
+          final String checkName = checkRunEvent.checkRun!.name!;
+          final RepositorySlug slug = checkRunEvent.repository!.slug();
+
+          // TODO: Can't access branch from [checkRunEvent.checkRun.checkSuite] because head_branch is not deserialized
+          // See https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=rerequested#check_run
+          final String gitBranch = Config.defaultBranch(slug);
+
+          // Only merged commits are mirrored. If a matching commit is found on GoB, this must be a postsubmit checkrun.
+          final GerritCommit? gobCommit = await gerritService.findMirroredCommit(slug, sha);
```

Comment:
If you're using the GoB mirror to determine if an item is merged, why not use the Datastore instead?

Reply:
So, Godofredo suggested this approach in https://github.com/flutter/flutter/issues/116739#issuecomment-1376592557. I believe this is in part to avoid mixing with the upcoming datastore refactor.

Reply:
Is there a doc for the proposed refactor? I presume we'd add a field that indicates if a task is presubmit vs postsubmit.

Otherwise, this adds a lot of extra complexity for now without much reason.

Reply:
Which event are we using to insert the commits into the datastore? do we only insert post-submit commits?

The database refactoring is a different project and I don't think there is a design doc yet.

Reply:
Cocoon processes commits through (1) GitHub webhook event for merged pull requests (2) /api/vacuum-github-commits cronjob for checking if the default branch has every commit. https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/service/scheduler.dart;l=123 is the only location inserting tasks to the datastore, and that only does it for merged commits.

Reply:
I suppose this could be updated to rely on the datastore after that new field is added in the refactor.

Reply:
Tasks today is only postsubmit, so relying on it SGTM